### PR TITLE
[MM-18121] Fix config set panic

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR will fix config set panic.
The bug was due to the fact that you can not convert map[string]someType to the map[string]interface{}
There is another issue here, namely, plugins that have dots in the name will not be able to set their settings. The issue will be addressed in separate PR.
#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-18121

### CODE REVIEW

File: cmd/mattermost/commands/config_test.go [hunk 1]
```diff
@@ -141,6 +141,12 @@ func TestConfigSet(t *testing.T) {
 		assert.NotContains(t, string(output), "invalid")
 	})
 
+	t.Run("Error when the parameter of an unknown plugin is set", func(t *testing.T) {
```
Comment (lieut-data) [inline cmd/mattermost/commands/config_test.go]: nit: indentation here seems off?

File: cmd/mattermost/commands/config.go [hunk 2]
```diff
@@ -308,6 +308,9 @@ func UpdateMap(configMap map[string]interface{}, configSettings []string, newVal
 		if len(configSettings) == 1 {
 			return errors.New("unable to set multiple settings at once")
 		}
+		if value.Len() == 0 {
+			return fmt.Errorf("unable to find a setting with that name %s", configSettings[1])
```
Comment (lieut-data) [inline cmd/mattermost/commands/config.go]: Language here might be hard to parse, perhaps:
```suggestion
			return fmt.Errorf("unable to find a setting named: %s", configSettings[1])
```

**Comment by lindalumitchell** [comment]
Also, are there QA test steps we can add to the JIRA ticket as needed?

**Comment by lindalumitchell** [comment]
Thanks! Test steps added to JIRA ticket, and unit test present. Removing QA Review label.
